### PR TITLE
Support Content-Transfer-Encoding binary

### LINKED ISSUE #1168: MultipartWriter does not support binary content encoding
## Long story short

There is no support of binary encoding in MultipartWriter
Some services require binary encoding. Seems that easy to implement with just returning unprocessed chunks.
## Expected behaviour

Supported binary encoding
## Actual behaviour

Not implemented encoding
https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/multipart.py#L858
## Steps to reproduce

Specify header 'Content-Transfer-Encoding' to 'binary' in MultiPartWriter

### PR DESCRIPTION
**Pull request by kxepal**
This fixes #1168

### CODE REVIEW

**Comment by asvetlov** [comment]
LGTM

File: aiohttp/multipart.py [hunk 1]
```diff
@@ -854,6 +856,8 @@ def _apply_content_transfer_encoding(self, stream):
         elif encoding == 'quoted-printable':
             for chunk in stream:
                 yield binascii.b2a_qp(chunk)
+        elif encoding == 'binary':
+            yield from stream
```
Comment (AleksandrIakhnev) [inline aiohttp/multipart.py:860]: Does not work for me (ClientRequestError: Can not write request body'), we need:

```
 for chunk in stream:
      yield chunk
```

Comment (kxepal) [reply aiohttp/multipart.py:860]: Hm...ok, but why? This should be roughly the same.

Comment (AleksandrIakhnev) [reply aiohttp/multipart.py:860]: Have you tried to make actual POST request? It is exception here https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/client_reqrep.py#L427  (got: class 'generator'  - seems pretty clear that is not the same return

Comment (kxepal) [reply aiohttp/multipart.py:860]: Well, basically whole logic here is based around generators. Ok, will try to make a better test then to catch that problem. Thanks for report!

Comment (AleksandrIakhnev) [reply aiohttp/multipart.py:860]: Thank you for quick response!

Comment (kxepal) [reply aiohttp/multipart.py:860]: @aamalev Could you share the code you use to reproduce that issue? Just to make sure that we're on the same wave here.

Comment (AleksandrIakhnev) [reply aiohttp/multipart.py:860]: https://gist.github.com/AleksandrIakhnev/6541a3b3d47a84fc55ffd4790b8bb1e1

Comment (kxepal) [reply aiohttp/multipart.py:860]: Great, thanks a lot.

**Comment by asvetlov** [comment]
@kxepal do you want to include the patch into aiohttp 1.0?
Since I've finished #1173 personally I have no show stoppers for next release.

**Comment by kxepal** [comment]
@asvetlov Yes, let's do that. I'll try to get to the code soon to close this one.

File: aiohttp/multipart.py (hunk 1 above)
Comment (kxepal) [reply aiohttp/multipart.py:860]: @AleksandrIakhnev Sorry, but I failed to reproduce your issue: script you provided works for me returning 405 response from Google. I also add [test](https://github.com/KeepSafe/aiohttp/pull/1169/files#diff-a5c957a9c2be1397119d89b7aea0a7efR182) about to make sure that it works right and expected.

Are you sure that nothing is missed there in your example?

Comment (AleksandrIakhnev) [reply aiohttp/multipart.py:860]: Sorry, seems that I've tested not on the latest version of aiohttp, after updating it returns 405 response.

**Comment by asvetlov** [comment]
Now the PR is ready, isn't it?

File: aiohttp/multipart.py (hunk 1 above)
Comment (kxepal) [reply aiohttp/multipart.py:860]: Great! (:

**Comment by kxepal** [comment]
@asvetlov Indeed it is.

**Comment by asvetlov** [comment]
Awesome!
Let me make a new release than.
Not all bugs are fixed and still there are improvements worth to be done but we've reached to the next milestone.
aiohttp is ready for 1.0 :)

**Comment by kxepal** [comment]
@asvetlov Great! Do you plan to cut the new release today?

**Comment by asvetlov** [comment]
Yes. We was waiting it for very long, at least couple weeks longer that I expected.

**Comment by fafhrd91** [comment]
@kxepal hey, i am playing with rust async stuff. would you be interested in help me with mime type parsing?

**Comment by kxepal** [comment]
@fafhrd91 I'm not very good with rust, but sure, ping me if you need a help - will try to make it.

**Comment by fafhrd91** [comment]
@kxepal i am not good either, i just started week ago :)
basically, i need mimetype streaming parser, and it could be just general library.
my code is here https://github.com/fafhrd91/simplehttp

**Comment by kxepal** [comment]
@fafhrd91 Ok, will check, thanks!

**Pull request merged**